context/app/static/js/hooks/useSearchData.js

```diff
@@ -7,12 +7,31 @@ function useSearchData(query) {
   const [searchData, setSearchData] = useState({});
   const [isLoading, setIsLoading] = useState(true);
 
+  const queryCurrent = {};
+  Object.assign(queryCurrent, query);
+  /* eslint-disable prettier/prettier */
+  if (!('bool' in queryCurrent)) {
+    queryCurrent.bool = {};
+  }
+  if (!('must_not' in queryCurrent.bool)) {
+    queryCurrent.bool.must_not = {};
+  }
+  if (!('exists' in queryCurrent.bool.must_not)) {
+    queryCurrent.bool.must_not.exists = {};
+  }
+  if (!('field' in queryCurrent.bool.must_not.exists)) {
+    queryCurrent.bool.must_not.exists.field = 'next_revision_uuid';
+  } else {
+    throw new Error(`Bug: bool.must_not.exists.field already set: ${query}`);
+  }
+  /* eslint-enable prettier/prettier */
```

Comment:
```suggestion
  /* eslint-disable prettier/prettier */
  if (!('bool'     in queryCurrent))                      queryCurrent.bool = {};
  if (!('must_not' in queryCurrent.bool))                 queryCurrent.bool.must_not = {};
  if (!('exists'   in queryCurrent.bool.must_not))        queryCurrent.bool.must_not.exists = {};
  if (!('field'    in queryCurrent.bool.must_not.exists)) queryCurrent.bool.must_not.exists.field = 'next_revision_uuid';
  else {
    throw new Error(`Bug: bool.must_not.exists.field already set: ${query}`);
  }
  /* eslint-enable prettier/prettier */
```

I had wanted to commit something more like this, but it got auto-formatted.
There are utils out there to set deep properties... are we already using one in our codebase?